src/Main.purs

```diff
@@ -0,0 +1,529 @@
+      JQuery.select "#column2" >>= JQuery.display
+      JQuery.select "#showjs_label" >>= JQuery.display
+      JQuery.select "#showjs" >>= JQuery.display
+    _ -> do
+      JQuery.select "#column1" >>= JQuery.display
+      JQuery.select "#column2" >>= JQuery.display
+      JQuery.select "#showjs_label" >>= JQuery.display
+      JQuery.select "#showjs" >>= JQuery.display
+
+-- | Get the backend name from whatever is selected in the menu.
+getBackendNameFromView :: forall eff. Eff (dom :: DOM | eff) String
+getBackendNameFromView =
+  fromMaybe "core" <$>
+    (JQuery.select "input[name=backend_inputs]"
+      >>= \jq -> JQuery.filter jq ":checked"
+      >>= JQuery.getValueMaybe)
+
+-- | Get the backend configuration from whatever is selected in the menu.
+getBackendConfigFromView :: forall eff. Eff (dom :: DOM | eff) BackendConfig
+getBackendConfigFromView = getBackendFromString <$> getBackendNameFromView
+
+-- | Read query string options and update the state accordingly
+loadOptions
+  :: forall eff
+   . BackendConfig
+  -> Eff ( console :: CONSOLE
+         , confirm :: CONFIRM
+         , dom :: DOM
+         , timer :: TIMER
+         | eff
+         ) Unit
+loadOptions bc = do
+  viewMode <- getQueryStringMaybe "view"
+  case viewMode of
+    Just viewMode_
+      | viewMode_ `elem` ["sidebyside", "code", "output"]
+      -> JQuery.select ("#view_" <> viewMode_) >>= JQuery.click
+    _ -> pure unit
+
+  showJs <- getQueryStringMaybe "js"
+  case showJs of
+    Just showJs_ ->
+      JQuery.select "input:checkbox[name=showjs]" >>= JQuery.setProp "checked" (showJs_ == "true")
+    _ -> pure unit
+
+  autoCompile <- getQueryStringMaybe "compile"
+  case autoCompile of
+    Just autoCompile_ ->
+      JQuery.select "input:checkbox[name=auto_compile]" >>= JQuery.setProp "checked" (autoCompile_ == "true")
+    _ -> pure unit
+
+  gist <- getQueryStringMaybe "gist"
+  case gist of
+    Just gist_ -> JQuery.select "#view_gist" >>= JQuery.attr { href: "https://gist.github.com/" <> gist_ }
+    Nothing -> JQuery.select "#view_gist_li" >>= JQuery.hide
+
+-- | Setup event listeners for the backend dropdown menu.
+setupBackendMenu
+  :: forall eff
+   . BackendConfig
+  -> Eff
+       ( dom :: DOM
+       , confirm :: CONFIRM
+       , console :: CONSOLE
+       , timer :: TIMER
+       | eff
+       )
+       Unit
+setupBackendMenu bc@(BackendConfig backend) = do
+  JQuery.select ("#backend_" <> backend.backend) >>= JQuery.attr { checked: "checked" }
+  JQuery.select "input[name=backend_inputs]" >>= JQuery.on "change" \e jq -> do
+    bc_@(BackendConfig newBackend) <- getBackendConfigFromView
+
+    ok <- window >>= confirm ("Replace your current code with the " <> newBackend.backend <> " backend sample code?")
+    if ok
+      then navigateTo ("?backend=" <> newBackend.backend)
+      else void $ setTimeout 1000 do
+             compile bc_
+             cacheCurrentCode
+    hideMenus
+
+getBackend :: Backend -> BackendConfig
```

Comment:
This should go in `Types.purs` imo.

Reply:
I moved all of this under API actually. To do what I wanted, it had to be that way to avoid a dependency cycle.